### ISSUE TITLE
(#117) - 400 Error being returned when the route is not available for a particular day of the week

### ISSUE THREAD

**Issue by avravinder** [rxtrax/tracking#80]
When we post a delivery on Wednesday that is tied to a route Which does not deliver on Wednesday then the response comes back as 400 error. 
it should give a agreed upon error code if 400 is generic to more that one error so Pharmaserv can display a valid messages and the pharmacist can then reschedule it to day that the route is active on.

**Comment by RyanNerd** [comment]
400 is a [standard http status response code](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_Error) for an invalid request. Returning a non-standard status response code is not recommended. The JSON payload in the missing=>invalid section has the details of the invalid request: route_unavailable.

If additional information in the invalid response is needed then this issue needs to be labeled as an enhancement.

**Comment by jasonmadrid** [comment]
@RyanNerd We discussed this on our call with McKesson today.  I'll update you on the plan later.

**Comment by RyanNerd** [comment]
Ok. Sorry if I spoke out of turn. I worked on the route available verification code and wanted to make sure we didn't introduce anything non-standard.

**Comment by brprashanth** [comment]
Jason - We are not asking you to change the 400 generic http response code. We are good with that.

What we explained in the calling and what we expect is the API should return an error code instead of just a missing fields  section. The missing fields sections is hard to parse programatically and display an intuitive user friendly message to the user. 
On top of this missing fields section - if the API returns an "additional" pre-defined validation code such as say 10001 - Missing Route, 10002 - Route not available for the date selected, it will be of great help.
